etna/experimental/prediction_intervals/empirical.py

```diff
@@ -0,0 +1,91 @@
+    """Estimate prediction intervals using values of historical residuals.
+
+    1. Compute matrix of residuals  :math:`r_{it} = |\hat y_{it} - y_{it}|` using k-fold backtest, where :math:`i` is fold index.
+
+    2. Estimate quantiles levels, that satisfy the provided coverage, for the corresponding residuals distributions.
+
+    3. Estimate quantiles for each timestamp using computed residuals and levels.
+
+    `Reference implementation <https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.forecasting.conformal.ConformalIntervals.html>`_.
+    """
+
+    def __init__(self, pipeline: BasePipeline, coverage: float = 0.95, cutoff_borders: bool = True, stride: int = 1):
+        """Initialize instance of ``EmpiricalPredictionIntervals`` with given parameters.
+
+        Parameters
+        ----------
+        pipeline:
+            Base pipeline or ensemble for prediction intervals estimation.
+        coverage:
+             Interval coverage. In literature this value maybe referred as ``1 - alpha``.
+        cutoff_borders:
+             Ensure that the forecast lies inside the prediction interval.
+        stride:
+            Number of points between folds.
+        """
+        if not (0 <= coverage <= 1):
+            raise ValueError("Parameter `coverage` must be non-negative number not greater than 1!")
+
+        if stride <= 0:
+            raise ValueError("Parameter `stride` must be positive!")
+
+        self.coverage = coverage
+        self.cutoff_borders = cutoff_borders
+        self.stride = stride
+
+        super().__init__(pipeline=pipeline)
+
+    def _forecast_prediction_interval(
+        self, ts: TSDataset, predictions: TSDataset, quantiles: Sequence[float], n_folds: int
+    ) -> TSDataset:
+        """Estimate and store prediction intervals.
+
+        Parameters
+        ----------
+        ts:
+            Dataset to forecast.
+        predictions:
+            Dataset with point predictions.
+        quantiles:
+            Levels of prediction distribution.
+        n_folds:
+            Number of folds to use in the backtest for prediction interval estimation.
+
+        Returns
+        -------
+        :
+            Dataset with predictions.
+        """
+        residuals = residuals_matrices(pipeline=self, ts=ts, n_folds=n_folds, stride=self.stride)
+
+        prediction_target = predictions[:, :, "target"]
+
+        levels = 0.5 + np.array([-0.5, 0.5]) * self.coverage
+        lower_quantile, upper_quantile = np.quantile(residuals, levels, axis=0)
+
+        # cutoffs to keep prediction inside interval
+        if self.cutoff_borders:
+            upper_quantile = np.maximum(upper_quantile, 0)
+            lower_quantile = np.minimum(lower_quantile, 0)
+
+        upper_border = prediction_target + upper_quantile
+        lower_border = prediction_target + lower_quantile
+
+        upper_border.rename({"target": "target_upper"}, inplace=True, axis=1)
```

Comment:
Can' we specify exact quantiles here?

Reply:
This estimation differs from quantiles in the library. It uses same assumptions as the conformal method and computes quantiles on folds dimmention.